src/conversion/model-to-view-converters.js

```diff
@@ -478,6 +385,63 @@ export function remove() {
 	};
 }
 
+/**
+ * Function factory, crates marker to virtual selection converter.
+ * For more information see {@link module:engine/conversion/buildmodelconverter~ModelConverterBuilder#toVirtualSelection}.
+ *
+ * @param {module:engine/conversion/buildmodelconverter~VirtualSelectionDescriptor|Function} selectionDescriptor
+ * @return {Function} Marker to virtual selection converter.
+ */
+export function markerToVirtualSelection( selectionDescriptor ) {
+	return ( evt, data, consumable, conversionApi ) =>	{
+		const descriptor = typeof selectionDescriptor == 'function' ?
+			selectionDescriptor( data, consumable, conversionApi ) :
+			selectionDescriptor;
+
+		if ( !descriptor ) {
+			return;
+		}
+
+		const addMarker = evt.name.split( ':' )[ 0 ] == 'addMarker';
+		const modelItem = data.item;
+
+		// Return if model item is not present. This happens when whole marker conversion is fired before converting
+		// children of marker's range.
+		if ( !modelItem ) {
+			return;
+		}
+
+		if ( modelItem.is( 'textProxy' ) ) {
+			const viewElement = virtualSelectionDescriptorToAttribute( descriptor );
+			const converter = addMarker ?
+				wrapItem( viewElement, eventName => eventName ) :
+				unwrapItem( viewElement, eventName => eventName );
+
+			converter( evt, data, consumable, conversionApi );
```

Comment:
I agree that we should not use "attribute" as a shortcut from "attribute element".

But my problem with this code is that when we use one converter functions factory in another it's very hard to understand how the code works. Especially when, like in this case, each of them wants to have this conversion working differently. Especially when, in fact, we don't need most of the code from the inside of `wrapItem` and `unwrapItem` in this case. We should use here the code which we need. In case, of having some complicated logic duplicated (like this condition https://github.com/ckeditor/ckeditor5-engine/blob/9f7e0a2c813a3a8f7c673c471d8caa3621200226/src/conversion/model-to-view-converters.js#L276-L279) we can export it to the separate function (what will improve readability anyway).